### ISSUE TITLE
Pattern variable is not recognized in AND_AND_Expression

### ISSUE THREAD

**Issue by snjeza** [eclipse-jdt/eclipse.jdt.core#1759]
The related issue - https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/2992
Steps to reproduce:

- create the following class
```
import java.util.ArrayList;
import java.util.List;
public class X {
    public void test() {
        Object obj = "test"; 
        List<String> values = new ArrayList<>();
        if (obj instanceof String str && values.stream().anyMatch(str::equalsIgnoreCase)) {
            System.out.println(str);
        }
    }
}
```
- JDT returns
```
The method anyMatch(java.util.function.Predicate<? super java.lang.String>) in the type Stream<String> is not applicable for the arguments (java.util.function.Predicate<? super java.lang.String>)
```
The following code works fine
```
import java.util.ArrayList;
import java.util.List;
public class X {
    public void test() {
        Object obj = "test"; 
        List<String> values = new ArrayList<>();
        if (obj instanceof String str) {
        	if (values.stream().anyMatch(str::equalsIgnoreCase)) {
        		System.out.println(str);
        	}
        }
    }
}
```

**Comment by snjeza** [comment]
> Any objections to merging https://github.com/eclipse-jdt/eclipse.jdt.core/pull/1768 in preference to https://github.com/eclipse-jdt/eclipse.jdt.core/pull/1761 ??

No, It is fine.

**Comment by snjeza** [comment]
Closed in favour of #1768

**Comment by srikanth-sankaran** [comment]
PR is still not merged in, reopening ...
